custom_components/spotcast/__init__.py

```diff
@@ -206,6 +207,9 @@ def start_casting(call: ha_core.ServiceCall):
             # remove ? from badly formatted URI
             uri = uri.split("?")[0]
 
+            if uri.startswith("https"):
```

Comment:
raise logger error on invalid url. This should be a specific error message different than the URI. The current process could lead people to believe URL are not trully compatible and force them to revert to URI, instead of fixing the url